mapping/sustainability.yaml

```
@@ -57,15 +57,120 @@
+                      "id": "1",
+                      "name": "Agency for Agricultural Development of Morocco"
+                    }
+                  },
+                  {
+                    "id": "3",
+                    "description": "Operation funding/budget source not specified"
+                  }
+                ]
+              }
+            ]
+          },
+          "parties": [
+            {
+              "id": "1",
+              "name": "Agency for Agricultural Development of Morocco",
+              "roles": [
+                "funder"
+              ]
+            }
+          ]
+        }
 -   id: '1.5'
     title: Budget for preparation, implementation and operation
     module: Economic and fiscal
     indicator: Economic viability
     disclosure format: Specify the allocated budget for preparation, implementation, operation. If no amount is allocated for each of the stages, select the option 'amount not allocated" (E.g. Preparation \[currency and amount\], Implementation\[currency and amount\], Operation \[currency and amount\]
-    mapping: ''
-    example: ''
+    mapping: |-
+        Project level:
+
+        1. Get the `BudgetBreakdowns` object in the `budget.budgetBreakdowns` array that represents the budget breakdown by stage. If none exists yet:
+
+        - Add a `BudgetBreakdowns` object to the `budget.budgetBreakdowns` array.
+
+        2. Set it's `.id` incrementally and it's `.description` to "Breakdown by stage".
+
+        3. Get the `BudgetBreakdown` object in the budget breakdowns object for each of the 3 stages. If none exists yet:
+
+        - Add 3 `BudgetBreakdown` objects to the `budget.budgetBreakdowns.budgetBreakdown` array. For each object:
+        - Set the `.id` incrementally.
+        - Set the `.description` to the stage name - "Preparation", "Implementation" or "Operation". If the funding source is not known for a stage note append "funding/budget amount not allocated" to the `.description`.
+
+        4. For each stage, map the amount and currency to the budget breakdown's `.amount`
+    example: |-
+        {
+          "budget": {
+            "budgetBreakdowns": [
+              {
+                "id": "1",
+                "description": "Breakdown by stage",
+                "budgetBreakdown": [
+                  {
+                    "id": "1",
+                    "description": "Preparation",
+                    "amount": {
+                      "amount": 350000,
+                      "currency": "USD"
+                    }
+                  },
+                  {
+                    "id": "2",
+                    "description": "Implementation",
+                    "amount": {
+                      "amount": 300000,
+                      "currency": "USD"
+                    }
+                  },
+                  {
+                    "id": "3",
+                    "description": "Operation funding/budget amount not allocated"
+                  }
```

See comment on previous example.

```suggestion
                  }
```